packages/core-dialog/core-dialog.js

```diff
@@ -1 +1,137 @@
+import {name, version} from './package.json'
+import {IS_BROWSER, queryAll, addEvent, dispatchEvent} from '../utils'
 
+const UUID = `data-${name}-${version}`.replace(/\W+/g, '-') // Strip invalid attribute characters
+const SUPPORT = IS_BROWSER && typeof window.HTMLDialogElement !== 'undefined'
+const PREVIOUS = `${UUID}-previous`
+const KEYS = {ESC: 27, TAB: 9}
+
+const FOCUSABLE_ELEMENTS = `
+  [tabindex]:not([disabled]),
+  a:not([disabled]),
+  button:not([disabled]),
+  input:not([disabled]),
+  select:not([disabled]),
+  textarea:not([disabled])`
+
+export default function dialog (dialogs, open) {
+  const options = typeof open === 'object' ? open : {open}
+
+  return queryAll(dialogs).forEach((dialog) => {
+    dialog.setAttribute(UUID, '')
+    dialog.setAttribute('role', 'dialog')
+    dialog.setAttribute('aria-modal', true)
+    dialog.setAttribute('aria-label', options.label || dialog.getAttribute('aria-label'))
+
+    toggleDialog(dialog, options.open)
+    return dialog
+  })
+}
+
+addEvent(UUID, 'click', (event) => {
+  // This functions handles if the user clicked on a open or close button.
+  // We need to do this loop in order for the close button to know which
+  // dialog it should close.
+  for (let el = event.target, isClose; el; el = el.parentElement) {
+    const action = el.getAttribute('data-core-dialog')
+    isClose = isClose || action === 'close'
+    if (isClose) el.hasAttribute(UUID) && dialog(el, false)
+    else if (action) dialog(action, true)
+  }
+})
+
+addEvent(UUID, 'keydown', (event) => {
+  if (event.keyCode === KEYS.TAB) keepFocus(event)
+  if (event.keyCode === KEYS.ESC && !event.defaultPrevented) {
+    const activeDialog = getTopLevelDialog()
+    if (activeDialog) {
+      event.preventDefault()
+      dialog(activeDialog, false)
+    }
+  }
+})
+
+const getZIndexOfElement = (element) =>
+  Number(window.getComputedStyle(element).getPropertyValue('z-index')) || 0
+
+// Find the last focused element before opening the dialog
+const getLastFocusedElement = () =>
+  queryAll(`[${PREVIOUS}]`).sort((a, b) =>
+    a.getAttribute(PREVIOUS) < b.getAttribute(PREVIOUS)
+  )[0] || document.body
+
+const getTopLevelDialog = () =>
+  queryAll(`[${UUID}][open]`).sort((a, b) =>
+    getZIndexOfElement(a) < getZIndexOfElement(b)
+  )[0]
+
+function toggleDialog (dialog, open) {
+  const last = getLastFocusedElement()
+  const isOpen = toggleOpen(dialog, open)
+  const lastIndex = Number(last.getAttribute(PREVIOUS)) || 0
+  const topZIndex = Math.max(...queryAll('*').map(getZIndexOfElement))
+  let backdrop = dialog.nextElementSibling
+
+  if (isOpen) {
+    const focusable = queryFocusable(dialog)[0]
+    document.activeElement.setAttribute(PREVIOUS, lastIndex + 1)
+    dialog.style.zIndex = topZIndex + 2
+    focusable && focusable.focus()
+
+    // Next element should be backdrop. If not, create it and insert it
+    if (!backdrop || backdrop.nodeName !== 'BACKDROP') {
```

Comment:
Unsure, but maybe move the creation of backdrop to line 73 and always ensure a next element = backdrop? also, `insertAdjacentElement` returns inserted element, so can write:
```
backdrop = dialog.insertAdjacentElement('afterend', document.createElement('backdrop'))
```
backdrop should get `hidden` property set 2 times, so animation can run: now insertAdjacentElement + removeAttribute('hidden') = no animation to run :)